### PR TITLE
[jsk_robot_startup] Enable input condition and added shutdown_unchecked

### PR DESCRIPTION
**Pull request by iory**
# What is this?

This is solution for `2A` https://github.com/jsk-ros-pkg/jsk_robot/pull/1561#issuecomment-1223862080
> 2-A) Modify [shutdown.py](https://github.com/jsk-ros-pkg/jsk_robot/pull/1426/files#diff-cf0bde4b686b024bc710bc3ca3c01e227134a66e783ccea5defe6a0c90eccc29) ,

I added `~input_condition` param to check `battery_state`.

### CODE REVIEW

File: jsk_robot_common/jsk_robot_startup/README.md [hunk 1]
```diff
@@ -187,6 +187,15 @@ This node shuts down or reboots the robot itself according to the rostopic. Note
 
   Input topic that trigger reboot
 
+* `shutdown_unchecked` (`std_msgs/Empty`)
```
Comment (k-okada) [inline jsk_robot_common/jsk_robot_startup/README.md:190]: Does this node shoutdown computer if we set `~input_condition` to `None`? if so , I think  we do not need shutdown_unchecked. Other part is very nice.

Comment (iory) [reply jsk_robot_common/jsk_robot_startup/README.md:190]: We may want to run shutdown when the battery is charged. For example, blackout.
So shutdown_unchecked (shutdown with check) and shutdown (force shutdown) are required.

Comment (k-okada) [reply jsk_robot_common/jsk_robot_startup/README.md:190]: 1) in that case, we can ssh and run `shutdown`. Of course if robot can understand it is black out or earthquack, fires, then it is different story
2) Even if you wan to force shoutdown, set `~input_condition` to `None` and send `shutdown`

Comment (iory) [reply jsk_robot_common/jsk_robot_startup/README.md]: I see. I understand it.
I modified the code to disable `shutdown_unchecked`.
https://github.com/jsk-ros-pkg/jsk_robot/pull/1571/commits/9d359f0a80e1fc5cbcb0b0c0b5fe8cbcdb9f6d48

File: jsk_robot_common/jsk_robot_startup/README.md [hunk 2]
```diff
@@ -197,6 +206,24 @@ This node shuts down or reboots the robot itself according to the rostopic. Note
 
   Command to reboot the system. You can specify the reboot command according to your system.
 
+* `~input_condition` (String, default: ``None``)
+
+  Returning bool value condition using the given Python expression. The Python expression can access any of the Python builtins plus: ``topic`` (the topic of the message), ``m`` (the message) and ``t`` (time of message).
```
[the remark this answers is not in the document]
Comment (k-okada) [reply jsk_robot_common/jsk_robot_startup/README.md:211]: I did not understand `Python builtins plus ..` when I first saw it...

may be 

The Python expression can access any of the Python builtin **functions** plus

or 

In addition to a Python builtin functions, you can use ``topic`` ...... in the Python expression.

???

**Review by k-okada — CHANGES_REQUESTED** [review]
thank you , please review the document again. You mey ask someone who is good at English to help you.

File: jsk_robot_common/jsk_robot_startup/README.md [hunk 3]
```diff
@@ -181,12 +181,21 @@ This node shuts down or reboots the robot itself according to the rostopic. Note
 
 * `shutdown` (`std_msgs/Empty`)
 
-  Input topic that trigger shutdown
+  Input topic that trigger shutdown.
+
+  If `~input_condition` is set, evaluated `~input_condition` is `True` and this node received this topic, shutdown will be executed.
+
+  Even if you want to force shutdown, set `~input_condition` to `None` and send `shutdown` topic.
```
[the remark this answers is not in the document]
Comment (iory) [reply jsk_robot_common/jsk_robot_startup/README.md]: I fixed it as follows.
```
If you want to force a shutdown in any case, set `~input_condition` to `None` and send `shutdown` topic.
```

File: jsk_robot_common/jsk_robot_startup/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (iory) [reply jsk_robot_common/jsk_robot_startup/README.md]: Thanks. I fixed it as follows.

```
Specify condition to run `~shutdown_command` even if shutdown topic is received. Use a Python expression that returns a bool value.

In addition to a Python builtin functions, you can use ``topic`` (the topic of the message), ``m`` (the message) and ``t`` (time of message).
```

File: jsk_robot_common/jsk_robot_startup/README.md [hunk 4]
```diff
@@ -197,6 +206,24 @@ This node shuts down or reboots the robot itself according to the rostopic. Note
...
+  ```
+  input_condition: m.data == 'hello'
+  ```
+
+  If you want to check the frame id of the header, you can do the following.
+
+  ```
+  input1_condition: m.header.frame_id in ['base', 'base_link']
+  ```
+
```
[the remark this answers is not in the document]
Comment (iory) [reply jsk_robot_common/jsk_robot_startup/README.md:225]: Thanks.
I added an example for a real fetch robot.
https://github.com/jsk-ros-pkg/jsk_robot/pull/1571/commits/808b9f3dad554476a48a71bd28b2f5e58945a601

File: jsk_robot_common/jsk_robot_startup/README.md [hunk 5]
```diff
@@ -197,6 +206,24 @@ This node shuts down or reboots the robot itself according to the rostopic. Note
 
   Command to reboot the system. You can specify the reboot command according to your system.
 
+* `~input_condition` (String, default: ``None``)
+
+  Returning bool value condition using the given Python expression. The Python expression can access any of the Python builtins plus: ``topic`` (the topic of the message), ``m`` (the message) and ``t`` (time of message).
+
+  For example, ``~input`` topic is ``std_msgs/String`` and if you want to check whether a sentence is a ``hello``, you can do the following.
```
[the remark this answers is not in the document]
Comment (iory) [reply jsk_robot_common/jsk_robot_startup/README.md:214]: Thanks. I added.
https://github.com/jsk-ros-pkg/jsk_robot/pull/1571/commits/808b9f3dad554476a48a71bd28b2f5e58945a601

**Pull request merged**